### PR TITLE
 Fix kubelet-to-apiserver connection checks on controller nodes not to fail in certain cases

### PR DESCRIPTION
**Pull request by mumoshu**
- check-worker-connection was failing when audit logging is disabled
- In addition to that, it was failing since K8S v1.8 due to the change in K8S that no audits are logged when missing an audit policy file

Follow-up for #991 and #996 
Fix for https://github.com/kubernetes-incubator/kube-aws/pull/991#commitcomment-25459898
Depends on #1014

### CODE REVIEW

File: core/controlplane/config/credential.go [hunk 1]
```diff
@@ -80,7 +80,10 @@ func EncryptedCredentialCacheFromRawCredential(raw *RawCredentialOnDisk, bytesEn
 }
 
 func RawCredentialFileFromPath(filePath string, defaultValue *string) (*RawCredentialOnDisk, error) {
-	if _, err := os.Stat(filePath); os.IsNotExist(err) && defaultValue != nil {
+	if _, err := os.Stat(filePath); os.IsNotExist(err) {
+		if defaultValue == nil {
+			return nil, fmt.Errorf("%s must exist. Please confirm that you have not deleted the file manually", filePath)
+		}
```
Comment (danielfm) [inline core/controlplane/config/credential.go]: It seems like the contents of #1014 got mixed in here as well.

File: core/controlplane/config/templates/cloud-config-controller [hunk 2]
```diff
@@ -2942,6 +2951,109 @@ write_files:
...
+        # so only log at the Metadata level.
+        - level: Metadata
+          resources:
+            - group: "" # core
+              resources: ["secrets", "configmaps"]
+            - group: authentication.k8s.io
+              resources: ["tokenreviews"]
+          omitStages:
+            - "RequestReceived"
+        # Get repsonses can be large; skip them.
```
Comment (danielfm) [inline core/controlplane/config/templates/cloud-config-controller]: Typo in `repsonses`.

File: core/controlplane/config/templates/cloud-config-controller [hunk 3]
```diff
@@ -2942,6 +2951,109 @@ write_files:
...
+        - level: Request
+          verbs: ["get", "list", "watch"]
+          resources: ${known_apis}
```
Comment (danielfm) [inline core/controlplane/config/templates/cloud-config-controller]: We are not substituting the `known_apis` variable, like it's done [here](https://github.com/kubernetes/kubernetes/blob/v1.8.3/cluster/gce/gci/configure-helper.sh#L497).

File: core/controlplane/config/templates/cloud-config-controller [hunk 4]
```diff
@@ -3092,20 +3204,32 @@ write_files:
...
 
       queryserver() {
         echo "Checking to see if workers are communicating with API server."
-        docker logs --tail 10 ${DOCKERIMAGE} |& grep -v 127.0.0.1 | awk 'BEGIN {C=1} /kubelet\// && $8 == "200" {C=0} END {exit C}'
+        auditlogs | grep -v 127.0.0.1 | awk 'BEGIN {C=1} /kubelet\// && $8 == "200" {C=0} END {exit C}'
+      }
+
+      auditlogs() {
+        if [ "$AUDIT_LOG_PATH" == "/dev/stdout" ]; then
+          docker logs --since 11s ${DOCKERIMAGE} |& cat
```
Comment (danielfm) [inline core/controlplane/config/templates/cloud-config-controller]: Is there a reason behind this magic number here (`11s`)?

Comment (mumoshu) [reply core/controlplane/config/templates/cloud-config-controller]: `queryserver` is called every 10+ seconds below. I wanted to gather logs for periods slightly longer than that not to drop any lines(hopefully)

File: core/controlplane/config/templates/cloud-config-controller (hunk 3 above)
Comment (mumoshu) [reply core/controlplane/config/templates/cloud-config-controller]: Thanks! Manually substituted them anyway 😄

File: core/controlplane/config/templates/cloud-config-controller (hunk 2 above)
Comment (mumoshu) [reply core/controlplane/config/templates/cloud-config-controller]: Good catch! Fixed.

File: core/controlplane/config/credential.go (hunk 1 above)
Comment (mumoshu) [reply core/controlplane/config/credential.go]: Thanks for pointing that out!
Sorry but I had no time to resolving possible conflicts and testing them in isolations.
Marking this to depend on #1014 anyway.

**Comment by mumoshu** [comment]
I've manually verified this to work by updating my cluster few times:

- From no auditing to auditing enabled while setting logpath of /dev/stdout
- And then to /var/log/apiserver-audit.log

**Comment by mumoshu** [comment]
@danielfm Thanks for reviewing!

**Pull request merged**